external/fv3viz/fv3viz/_plot_cube.py

```diff
@@ -285,12 +292,36 @@ def pcolormesh_cube(
     kwargs["vmin"] = kwargs.get("vmin", np.nanmin(array))
     kwargs["vmax"] = kwargs.get("vmax", np.nanmax(array))
 
+    handles = []
     for tile in range(array.shape[0]):
         x = center_longitudes(lon[tile, :, :], central_longitude)
         y = lat[tile, :, :]
         for x_plot, y_plot, array_plot in _segment_plot_inputs(x, y, array[tile, :, :]):
-            p_handle = ax.pcolormesh(x_plot, y_plot, array_plot, **kwargs)
-    return p_handle
+            handles.append(ax.pcolormesh(x_plot, y_plot, array_plot, **kwargs))
+    return handles
+
+
+class UpdateablePColormesh:
```

Comment:
> is there a way it could be generalized to take in any plotting function that returns pcolormesh handles?

Yes and no. If I remember correctly, there are slightly different APIs for in-place updates of the data on different plotting routines. But apart from that, a big part of the handling for pcolormesh_cube is handling the fact that it actually performs a series of `.pcolormesh` calls on subsets of the data (to avoid any plotting commands that wrap around the domain boundaries). For simpler plotting commands that aren't using cartopy, we don't need extra tools at all.

There's probably a way to collect all those concerns into a class that supports some defined set of plotting routines with elegantly de-duplicated code, and I'd be all for extending into that direction.